tools/cosmovisor/cmd/cosmovisor/init.go

```diff
@@ -9,20 +9,28 @@ import (
 	"time"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 
 	"cosmossdk.io/log"
 	"cosmossdk.io/tools/cosmovisor"
 	"cosmossdk.io/x/upgrade/plan"
 )
 
-var initCmd = &cobra.Command{
-	Use:          "init <path to executable>",
-	Short:        "Initialize a cosmovisor daemon home directory.",
-	Args:         cobra.ExactArgs(1),
-	SilenceUsage: true,
-	RunE: func(cmd *cobra.Command, args []string) error {
-		return InitializeCosmovisor(nil, args)
-	},
+func NewIntCmd() *cobra.Command {
+	initCmd := &cobra.Command{
+		Use:          "init <path to executable>",
+		Short:        "Initialize a cosmovisor daemon home directory.",
+		Args:         cobra.ExactArgs(1),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			mustNoError(viper.BindPFlags(cmd.Flags()))
+			return InitializeCosmovisor(nil, args)
+		},
+	}
+
+	initCmd.Flags().String(cosmovisor.FlagExportConfig, "", "path to export the configuration file to (default is <-home_path->/cosmovisor/config.toml")
```

Comment:
I don't see where the default is set.
IMHO it shouldn't be current dir but within the node directory (DAEMON_HOME)

Reply:
The default is being set [here](https://github.com/cosmos/cosmos-sdk/pull/19995/files/74c8b24dfafd903bffb3e687f65a7803d3b48736#diff-56ac42a1b13fe0937bbfea75fb1048050db3792a6ea69ff873da1924a666bc3eR583).

I agree with you that's why I didn't set default path directly in flag because it depends on `DEAMON_HOME` so it has to be loaded at run time.

The default path is: `DEAMON_HOME/COSMOVISOR/config.toml`, you can check [here](https://github.com/cosmos/cosmos-sdk/pull/19995/files/74c8b24dfafd903bffb3e687f65a7803d3b48736#diff-56ac42a1b13fe0937bbfea75fb1048050db3792a6ea69ff873da1924a666bc3eR84).

Reply:
I see it, thanks!